plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -489,6 +529,14 @@ public static LongWriteFunction timestampWriteFunction(TimestampType timestampTy
         return (statement, index, value) -> statement.setObject(index, fromPrestoTimestamp(value));
     }
 
+    public static ObjectWriteFunction longTimestampWriteFunction(TimestampType timestampType)
+    {
+        checkArgument(timestampType.getPrecision() > MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
```

Comment:
add
```
checkArgument(timestampType.getPrecision() <= 9
```

since otherwise the function is not useful, as it goes via LDT

Reply:
Done!